File: 226. Invert Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        TreeNode tmpl = root.left;
        TreeNode tmpr = root.right;
        root.left = tmpr;
        root.right = tmpl;
         
        invertTree(root.left);
        invertTree(root.right);

        return root;
    }
}

File: 226. Invert Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        def invert(root):
            if not root:
                return root
            tmp = root.left
            root.left = root.right
            root.right = tmp
            invert(root.left)
            invert(root.right)
            
            return root
        
        return invert(root)
